h2o-algos/src/main/java/hex/tree/dt/binning/BinningStrategy.java

```diff
@@ -38,51 +37,63 @@ public enum BinningStrategy {
             return roundToNDecimalPoints(number, DECIMALS_TO_CONSIDER);
         }
 
-        private List<Bin> createEmptyBinsFromBinningValues(List<Double> binningValues, double realMin, double realMax) {
-            List<Bin> emptyBins = new ArrayList<>();
+        private List<AbstractBin> createEmptyBinsFromBinningValues(List<Double> binningValues, double realMin, double realMax) {
+            List<AbstractBin> emptyBins = new ArrayList<>();
             // create bins between nearest binning values, don't create bin starting with the last value (on index size - 1)
             for (int i = 0; i < binningValues.size() - 1; i++) {
                 emptyBins.add(
-                        new Bin(roundToNDecimalPoints(binningValues.get(i)),
+                        new NumericBin(roundToNDecimalPoints(binningValues.get(i)),
                                 roundToNDecimalPoints(binningValues.get(i + 1))));
             }
             // set the firs min to some lower value (relative to step) so the actual value equal to min is not lost
-            emptyBins.get(0)._min = realMin - 0.0001 * (binningValues.get(1) - binningValues.get(0));
+            ((NumericBin) emptyBins.get(0))._min = realMin - 0.0001 * (binningValues.get(1) - binningValues.get(0));
             // set the last max to the real max value to avoid precision troubles
-            emptyBins.get(emptyBins.size() - 1)._max = realMax;
+            ((NumericBin) emptyBins.get(emptyBins.size() - 1))._max = realMax;
```

Comment:
I am not satisfied with this casting to `NumericBin`. It should be safe to make the list type of `NumricBin`  (like `List<AbstractBin> emptyBins = new ArrayList<>();`) or am I missing something?